### PR TITLE
Feat/add trybuild crate as proc macro tester

### PR DESCRIPTION
**Pull request by Kilerd**
this PR is to add the trybuild as the proc macro tester,  and the folder `tests` is for persisting the struct we used to test macro during development.

### CODE REVIEW

**Comment by Kilerd** [comment]
I've update the commit of cargo fmt

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /target
 /.idea
 Cargo.lock
+db_path
```
Comment (ethe) [inline .gitignore:4]: Please add a new line here

Comment (Kilerd) [reply .gitignore]: done

**Review by ethe — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (ethe) [inline .gitignore:4]: I think we should clean every temporary files created by test cases, maybe we should create another issue.

**Comment by ethe** [comment]
Thank you! I wish there are more collaborations in the future 😄

**Pull request merged**